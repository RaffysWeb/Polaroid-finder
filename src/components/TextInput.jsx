import React from 'react';
import PropTypes from 'prop-types';

const TextInput = ({ placeholder, value, onChange, name, forwardedRef }) => (
  <input
    type="text"
    value={value}
    onChange={onChange}
    name={name}
    placeholder={placeholder}
    ref={forwardedRef}
  />
);

TextInput.defaultProps = {
  value: ''
};

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  placeholder: PropTypes.string.isRequired,
  forwardedRef: PropTypes.instanceOf(Object).isRequired,
  onChange: PropTypes.func.isRequired
};

export default TextInput;
