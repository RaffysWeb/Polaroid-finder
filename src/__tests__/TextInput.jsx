import React from 'react';
import { create } from 'react-test-renderer';
import TextInput from '../components/TextInput';

describe('TextInput tests', () => {
  test('TextInput it matches the snapshot', () => {
    const component = create(<TextInput name="test" onChange={jest.fn()} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
