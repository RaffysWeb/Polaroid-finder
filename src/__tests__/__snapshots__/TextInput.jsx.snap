// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput tests TextInput it matches the snapshot 1`] = `
<input
  className="sc-bdVaJa kTNRlN"
  name="test"
  onChange={[MockFunction]}
  placeholder=""
  type="text"
  value=""
/>
`;
